packages/react-native-sdk/docusaurus/docs/reactnative/06-advanced/11-native-permissions.mdx

```diff
@@ -0,0 +1,180 @@
+
+```tsx
+import { useEffect } from 'react';
+import { Platform } from 'react-native';
+// We will be using react-native-permissions lib. to manage our permissions
+import {
+  checkMultiple,
+  PERMISSIONS,
+  PermissionStatus,
+  requestMultiple,
+  RESULTS,
+} from 'react-native-permissions';
+
+import { StreamVideoRN } from '@stream-io/video-react-native-sdk';
+
+export const useSyncPermissions = () => {
+  useEffect(() => {
+    // Check, update and sync permissions after parent component mounts
+    requestAndUpdatePermissions();
+  }, []);
+  // useAppStateListener(handleOnForeground, () => {});
+};
+
+const requestAndUpdatePermissions = async () => {
+  if (Platform.OS === 'ios') {
+    // Request camera and mic permissions on iOS
+    const results = await requestMultiple([
+      PERMISSIONS.IOS.CAMERA,
+      PERMISSIONS.IOS.MICROPHONE,
+    ]);
+    // Sync the permissions with the Stream Video SDK
+    // This ensures that the SDK will subscribe to audio/video devices as preparation for a call.
+    iOSProcessResultsAndSetToConfig(results);
+    return;
+  } else if (Platform.OS === 'android') {
+    // Request camera, mic, bluetooth and notification permissions on Android
+    const results = await requestMultiple([
+      PERMISSIONS.ANDROID.CAMERA,
+      PERMISSIONS.ANDROID.RECORD_AUDIO,
+      PERMISSIONS.ANDROID.BLUETOOTH_CONNECT,
+      PERMISSIONS.ANDROID.POST_NOTIFICATIONS,
+    ]);
+    // Sync the camera and permissions with the Stream Video SDK
+    // This ensures that the SDK will subscribe to audio/video devices as preparation for a call.
+    androidProcessResultsAndSetToConfig(results);
+  }
+};
+
+const androidProcessResultsAndSetToConfig = (
+  results: Record<
+    'android.permission.CAMERA' | 'android.permission.RECORD_AUDIO',
+    PermissionStatus
+  >,
+) =>
+  // Sync the permissions with the Stream Video SDK
+  // by setting the permissions to the config.
+  StreamVideoRN.setPermissions({
+    isCameraPermissionGranted:
+      results[PERMISSIONS.ANDROID.CAMERA] === RESULTS.GRANTED,
+    isMicPermissionGranted:
+      results[PERMISSIONS.ANDROID.RECORD_AUDIO] === RESULTS.GRANTED,
+  });
+
+const iOSProcessResultsAndSetToConfig = (
+  results: Record<
+    'ios.permission.CAMERA' | 'ios.permission.MICROPHONE',
+    PermissionStatus
+  >,
+) =>
+  // Sync the permissions with the Stream Video SDK
+  // by setting the permissions to the config.
+  StreamVideoRN.setPermissions({
+    isCameraPermissionGranted:
+      results[PERMISSIONS.IOS.CAMERA] === RESULTS.GRANTED,
+    isMicPermissionGranted:
+      results[PERMISSIONS.IOS.MICROPHONE] === RESULTS.GRANTED,
+  });
+```
+
+## Sync Permissions on App State Change
+In order to sync the permissions when the app comes back to foreground, we will add a listener to the app state change event.
+This will ensure that the permissions are always in sync when the user comes back to the app from background after
+changing the permissions in the app's settings screen.
```

Comment:
```suggestion
## Sync Permissions on App State Change

In order to sync the permissions when the app comes back to the foreground, we will add a listener to the app state change event.
This will ensure that the permissions are always in sync when the user comes back to the app from the background after changing the permissions in the app's settings screen.
```